### PR TITLE
Add error reporting examples to rn-tester turbo modules

### PR DESCRIPTION
**Pull request by krystofwoldrich**
## Summary:

This PR is adding examples of Turbo Modules functions throwing runtime exceptions and asserts. This should make it easier to collaborate and develop the error reporting for a new architecture that is being discussed in the React Native New Architecture Working Group -> https://github.com/reactwg/react-native-new-architecture/discussions/122.

I'm not sure what return type should be used for the JS function returning `Promise<void>` in Cxx, I used [`AsyncPromise<jsi::Value>`](https://github.com/facebook/react-native/pull/36729/files#diff-9cebc75f48fd35fd6fef71138f98dfd0ba28a754b2aab0d6fe44fd685f74ce16R135), what would you use, I've not found `void` type to use?

### Added functions

The table shows the current behavior. 

<table>
<tr>
	<td> Function
	<td> Description
	<td> Turbo Module
	<td> Cxx Module
<tr>
	<td> voidFuncThrows
	<td> function with return type void throws a runtime exception
	<td> platform error no JS stack trace
	<td> JS error no native stack trace
<tr>
	<td> getObjectThrows
	<td> function with return type object throws a runtime exception
	<td> JS error no platform stack trace 
	<td> JS error no native stack trace
<tr>
	<td> promiseThrows
	<td> function with return type promise throws a runtime exception before settling the promise
	<td> platform error no JS stack trace
	<td> JS error no native stack trace
<tr>
	<td> voidFuncAssert
	<td> function with return type void asserts
	<td> platform error no JS stack trace
	<td> native error no JS stack trace
<tr>
	<td> getObjectAssert
	<td> function with return type object asserts 
	<td> JS error no platform stack trace
	<td> native error no JS stack trace
<tr>
	<td> promiseAssert
	<td> function with return type promise asserts before settling the promise
	<td> platform error no JS stack trace
	<td> native error no JS stack trace
</table>

## Changelog:

[INTERNAL] [ADDED] - Error reporting examples in rn-tester turbo modules

## Test Plan:

This PR doesn't change any RN behavior. Only shows the current state by adding an example to rn-tester.

I'm happy to add these examples to the unit/integration test, just point me to where would be a good place.

### CODE REVIEW

File: packages/rn-tester/js/examples/TurboModule/NativeCxxModuleExampleExample.js [hunk 1]
```diff
@@ -155,9 +203,6 @@ class NativeCxxModuleExampleExample extends React.Component<{||}, State> {
         'Cannot load this example because TurboModule is not configured.',
       );
     }
-    Object.keys(this._tests).forEach(item =>
-      this._setResult(item, this._tests[item]()),
-    );
```
Comment (javache) [inline packages/rn-tester/js/examples/TurboModule/NativeCxxModuleExampleExample.js:160]: Can we do this for all platforms now?

**Comment by javache** [comment]
Can you summarize the current behaviour / add a video of the screen you added? That will be helpful to understand future changes.

File: packages/rn-tester/js/examples/TurboModule/NativeCxxModuleExampleExample.js (hunk 1 above)
Comment (krystofwoldrich) [reply packages/rn-tester/js/examples/TurboModule/NativeCxxModuleExampleExample.js:160]: I'm not sure what you mean.

I've removed the autorun of the tests because some of the crash tests can't be watched in JS at the moment and the app would crash immediately.

**Comment by krystofwoldrich** [comment]
@javache Thank you for the feedback.
 I've added a table with the functions overview and I can add a video too.

**Comment by cortinico** [comment]
Thanks for sending this over @krystofwoldrich, before we can import this we'll have to make sure CircleCI is all green 👍

**Comment by krystofwoldrich** [comment]
@cortinico Yes, for sure, I didn't have Flow set up locally, so I missed some of the errors that showed in the CI.
